sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/IdempotentAmqpEventBatch.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
+using Azure.Messaging.EventHubs.Producer;
+using Microsoft.Azure.Amqp;
+
+namespace Azure.Messaging.EventHubs.Amqp
+{
+    /// <summary>
+    ///   A set of events with known size constraints, based on messages to be sent
+    ///   using an AMQP-based transport appropriate for idempotent publishing.
+    /// </summary>
+    ///
+    internal class IdempotentAmqpEventBatch : AmqpTransportEventBatch
```

Comment:
would it make sense to seal?

Reply:
Can't do it in this case;  the model factory implementation is extending this.  😄